src/dropboxHelper.js

```diff
@@ -0,0 +1,121 @@
+// Functions for storing notes to dropbox
+import { pathJoin, readBlobAsString } from './utils';
+import { dropboxConfig } from './userConfig';
+import { noteNameFromIdAndTitle } from './NoteData';
+
+const NOTES_FOLDER_PATH = pathJoin(
+    dropboxConfig.notesFolderParent === '/' ? '' : dropboxConfig.notesFolderParent,
+    dropboxConfig.notesFolderName
+);
+
+// Helper class for saving and retrieving notes from dropbox.
+class DropboxHelper {
+    // TODO: Remove this ctor altogether
+    constructor(notesFolderPath = NOTES_FOLDER_PATH) {
+        this.dbx = undefined;
+        this.notesFolderPath = notesFolderPath;
+    }
+
+    // Will be initialized after we complete setting up dropbox.
+    init(dbx) {
+        console.assert(!!dbx);
+        this.dbx = dbx;
+    }
+
+    isInitialized() {
+        return !!this.dbx;
+    }
+
+    // Save given editor value. We currently store only the JSON representation of the note.
+    // TODO: store plain text representation alongside it.
+    async save(noteData) {
+        if (!this.dbx) {
+            console.log('dropbox was not initialized. so not saving.');
+            return undefined;
+        }
+
+        if (!noteData.videoId || !noteData.videoTitle) {
+            console.warn('No videoId or videoTitle given');
+            return undefined;
+        }
+
+        const fileName = noteNameFromIdAndTitle(noteData.videoId, noteData.videoTitle);
```

Comment:
A video title might change (a user might edit video's title). Better to have path based on something thats immutable.